lib/eventFactory.js

```diff
@@ -314,43 +252,37 @@ class EventFactory extends BaseFactory {
                         modelConfig.commit = commit;
                         modelConfig.createTime = (new Date()).toISOString();
 
-                        if (config.workflowGraph) {
-                            modelConfig.workflowGraph = config.workflowGraph;
-
-                            return modelConfig;
-                        }
-
                         // Get latest config
                         return getLatestConfig({
                             pipeline,
                             eventConfig: config
-                        }).then((latestConfig) => {
-                            if (modelConfig.type === 'pr') {
-                                return latestConfig;
-                            }
+                        });
+                    })
+                    .then((latestConfig) => {
+                        if (config.parentEventId) {
+                            modelConfig.parentEventId = config.parentEventId;
+                            modelConfig.workflowGraph = config.workflowGraph;
```

Comment:
Don't need this line any more. We will get the `workflowGraph` form latestConfig.

Reply:
We still need this for the restart of event, so it will get the workflowGraph from the parent event.

Reply:
Cool cool, messed this with my PR. Once we have `pipeline.sync(ref)`, we can remove this. 👍